app/workers/teacher_training_public_api/sync_provider_worker.rb

```diff
@@ -0,0 +1,17 @@
+module TeacherTrainingPublicAPI
+  class SyncProviderWorker
+    include Sidekiq::Worker
+    sidekiq_options retry: 3, queue: :low_priority
+
+    def perform(provider_code)
+      provider_from_api = Provider.fetch(provider_code)
+
+      if provider_from_api.present?
+        TeacherTrainingPublicAPI::SyncProvider.new(
```

Comment:
Maybe we can set `run_in_background` to false to avoid kicking off another background job and finishing everything in the current job instead?

Reply:
Good question. As that value trickles all the way down to `SyncSites` I think it's probably better not to lose everything if one of those many HTTP requests fails or gets stuck. So I think I'd prefer to keep it async?